lib/emque/consuming/adapters/rabbit_mq/delayed_message_worker.rb

```diff
@@ -0,0 +1,114 @@
+                  "x-delayed-type" => "direct",
+                }
+              }
+            )
+
+            self.queue = channel.queue(
+              "emque.#{config.app_name}.delayed_message",
+              :durable => config.adapter.options[:durable],
+              :auto_delete => config.adapter.options[:auto_delete],
+              :arguments => {
+                "x-dead-letter-exchange" => "#{config.app_name}.error"
+              }
+            ).bind(delayed_message_exchange)
+          end
+
+          def start
+            logger.info "#{log_prefix} starting..."
+            queue.subscribe(:manual_ack => true, &method(:process_message))
+            logger.debug "#{log_prefix} started"
+          end
+
+          def stop
+            logger.debug "#{log_prefix} stopping..."
+            super do
+              logger.debug "#{log_prefix} closing channel"
+              channel.close
+            end
+            logger.debug "#{log_prefix} stopped"
+          end
+
+          private
+
+          attr_accessor :channel, :delayed_message_exchange, :queue
+
+          def process_message(delivery_info, metadata, payload)
+            begin
+              logger.info "#{log_prefix} processing message #{metadata}"
+              logger.debug "#{log_prefix} payload #{payload}"
+              message = Emque::Consuming::Message.new(
+                :original => payload
+              )
+              ::Emque::Consuming::Consumer.new.consume(:process, message)
+              channel.ack(delivery_info.delivery_tag)
+            rescue StandardError => ex
+              if retryable_errors.any? { |error| ex.class.to_s =~ /#{error}/ }
+                retry_errors(delivery_info, metadata, payload, ex)
+              else
+                channel.nack(delivery_info.delivery_tag)
+              end
+            end
+          end
+
+          def retry_errors(delivery_info, metadata, payload, ex)
+            headers = metadata[:headers] || {}
+            retry_count = headers.fetch("x-retry-count", 0)
+
+            if retry_count <= retryable_error_limit
+              logger.info("Retrying Retryable Error #{ex.class}, with count #{retry_count}")
+              headers["x-retry-count"] = retry_count + 1
+              headers["x-delay"] = delay_ms_time(retry_count)
+              channel.ack(delivery_info.delivery_tag)
+              delayed_message_exchange.publish(payload, {:headers =>headers})
+            else
+              logger.info("Retryable Error: #{ex.class} ran out of retries at #{retry_count}")
+              channel.nack(delivery_info.delivery_tag)
+            end
+          end
+
+          def retryable_errors
+            config.retryable_errors
+          end
+
+          def retryable_error_limit
+            config.retryable_error_limit
+          end
+
+          def log_prefix
+            "RabbitMQ DelayedMessageWorker:"
+          end
+
+          def delay_ms_time(retry_count)
```

Comment:
this may be a good place to start testing? with a spec running through multiple times to test a range of back-offs?